dialoptions.go

```diff
@@ -89,6 +96,14 @@ type DialOption interface {
 
 var globalDialOptions []DialOption
 
+// LateApplyDialOption takes a parsed target and returns a dial option to apply.
+type LateApplyDialOption interface {
```

Comment:
Unexport?  Move to `internal`?  I really don't want users seeing this hack.

Reply:
Will need to move to internal/ since the csm/ package needs to typecast the any in internal. But yeah makes sense I didn't want this to be in exported namespace (at least right now the user can't set it :)) but forgot I can just put in internal/.

Reply:
Discussed further offline; unexported and typecast from any in SetGlobalDialOptions.